src/constraints/outage_constraints.jl

```diff
@@ -276,77 +286,95 @@ end
     # and can't charge the battery unless binMGStorageUsed = 1
     @constraint(m, [s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
-        m[:dvStoragePower]["ElectricStorage"] >= sum(m[:dvMGProductionToStorage][t, s, tz, ts] for t in p.techs.elec)
+        m[:dvStoragePower]["ElectricStorage"] >= sum(m[:dvMGProductionToStorage]["ElectricStorage", t, s, tz, ts] for t in p.techs.elec)
     )
     
     # Dispatch from MG storage is no greater than inverter capacity
     # and can't discharge from storage unless binMGStorageUsed = 1
     @constraint(m, [s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
-        m[:dvStoragePower]["ElectricStorage"] >= m[:dvMGDischargeFromStorage][s, tz, ts]
+        m[:dvStoragePower]["ElectricStorage"] >= m[:dvMGDischargeFromStorage]["ElectricStorage", s, tz, ts] 
     )
     
     # Dispatch to and from electrical storage is no greater than power capacity
     @constraint(m, [s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
-        m[:dvStoragePower]["ElectricStorage"] >= m[:dvMGDischargeFromStorage][s, tz, ts]
-            + sum(m[:dvMGProductionToStorage][t, s, tz, ts] for t in p.techs.elec)
+        m[:dvStoragePower]["ElectricStorage"] >= m[:dvMGDischargeFromStorage]["ElectricStorage", s, tz, ts] 
+            + sum(m[:dvMGProductionToStorage]["ElectricStorage", t, s, tz, ts] for t in p.techs.elec)
+    )
+
+    # Discharge from electric storage
+    @constraint(m, [b in p.s.storage.types.elec, s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
+        m[:dvMGDischargeFromStorage][b, s, tz, ts] >= m[:dvMGStorageToElectrolyzer][b, s, tz, ts]
+            + m[:dvMGStorageToCompressor][b, s, tz, ts]
     )
     
     # State of charge upper bound is storage system size
     @constraint(m, [s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
-        m[:dvStorageEnergy]["ElectricStorage"] >= m[:dvMGStoredEnergy][s, tz, ts]
+        m[:dvStorageEnergy]["ElectricStorage"] >= m[:dvMGStoredEnergy]["ElectricStorage", s, tz, ts]
     )
     
     if solver_is_compatible_with_indicator_constraints(p.s.settings.solver_name)
         @constraint(m, [s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
-            !m[:binMGStorageUsed] => { sum(m[:dvMGProductionToStorage][t, s, tz, ts] for t in p.techs.elec) <= 0 }
+            !m[:binMGStorageUsed] => { sum(m[:dvMGProductionToStorage]["ElectricStorage", t, s, tz, ts] for t in p.techs.elec) <= 0 }
```

Comment:
if we aren't going to have an equivalent bin var for h2, we should document that